sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -339,10 +340,15 @@ private[hive] class HiveClientImpl(
 
   override def getDatabase(dbName: String): CatalogDatabase = withHiveState {
     Option(client.getDatabase(dbName)).map { d =>
+      // default database's location always use the warehouse path
+      val dbLocation = if (dbName == SessionCatalog.DEFAULT_DATABASE) {
```

Comment:
sorry, I don't get the point, if this logic is exactly what we expected, we'd better to replace it at the beginning?